build/generate-style-spec.ts

```diff
@@ -17,31 +17,51 @@ function propertyType(property) {
     }
 
     const baseType = (() => {
-        switch (property.type) {
-            case 'string':
-            case 'number':
-            case 'boolean':
-                return property.type;
-            case 'enum':
-                return unionType(property.values);
-            case 'array': {
-                const elementType = propertyType(typeof property.value === 'string' ? {type: property.value, values: property.values} : property.value);
-                if (property.length) {
-                    return `[${Array(property.length).fill(elementType).join(', ')}]`;
-                } else {
-                    return `Array<${elementType}>`;
+        if (typeof property.type === 'string') {
```

Comment:
Isn't this always string?
Can you elaborate on what the generation output changes are here?

Reply:
i was just confused by the `type`, because of projection.`type` = `mercator`, and `type`: `projection` and `type`: `emum`